cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/service/http/BasicHttpServiceContext.java

```diff
@@ -253,4 +258,23 @@ public void releaseCallResources() {
 
     return evaluated;
   }
+
+  /**
+   * Get preferences for the supplied namespace.
+   *
+   * If the namespace doesn't exist, this method returns null.
+   *
+   * @param namespace the name of the namespace to fetch preferences for.
+   * @param resolved true if resolved properties are desired.
+   * @return Map containing the preferences for this namespace
+   * @throws IOException if the preferencesFetcher could not complete the request.
+   */
+  @Override
+  public Map<String, String> getPreferencesForNamespace(String namespace, boolean resolved) throws IOException {
+    try {
+      return preferencesFetcher.get(new NamespaceId(namespace), resolved).getProperties();
+    } catch (NotFoundException nfe) {
+      return null;
```

Comment:
Don't return `null`. Use empty map instead.

Reply:
or if you need to signal the namespace is not found, throws an appropriate exception to indicate that.

Reply:
I discussed this with @albertshau last week. The NotFoundException class lives in `cdap-common` which is not available in the `cdap-system-app-api` module.

The alternative here is to re-implement an exception class to wrap this exception, or return null.

We already use `null` as a way to stop exception propagation in the `cdap-app-fabric` package, and as I discussed with Albert, `null` is not really ambiguous as querying for properties for an existing namespace will always return a map (that could be empty). 

If you feel strongly about this, I'll implement a new `NamespaceNotFound` exception in the `cdap-system-app-api` to wrap this `NotFoundException`.

Reply:
Generally I would not have `null` return at all, especially as a public API contract, since it is error-prone on the caller side and could easily generate mis-leading logs.

If you don't want to port the exception class, you can also just throw a IllegalArgumentException with a proper message saying that the namespace doesn't exist. At least if someone is forgetting to catch the exception, it still would generate a more meaningful message than "null".

Reply:
I think using `IllegalArgumentException` is a good idea, let me tweak the code.